### ISSUE TITLE
Clean up usages of createObservableArray.getArray

### ISSUE THREAD

**Issue by samreid** [phetsims/axon#279]
As described in https://github.com/phetsims/axon/issues/266#issuecomment-575941906

I improved documentation in the commits.  I noticed there are 86 occurrences of `.getArray()` across the codebase.  Many of these could be converted to use ObservableArray.filter and ObservableArray.map which were improved in https://github.com/phetsims/axon/issues/265 or other ObservableArray methods.  Others may have simple solutions, and the ones that remain should be documented accordingly as prescribed in https://github.com/phetsims/axon/issues/266#issuecomment-540131119.  For example, density and buoyancy has:

```js
return _.find( this.masses.getArray(), mass => mass.isBoat() ) || null;
```

But this could likely be changed to something like:
```js
return this.masses.find( mass => mass.isBoat() ) || null;
```

I'm not sure who should work on this or when it should be done.  Should this be a chip away issue, so each developer can edit and test their own sims?  @ariel-phet what do you recommend?

**Comment by samreid** [comment]
In the preceding commit, I implemented one of the conversions.  I first ran the code to make sure worked correctly, and that I could understand when it was being called in the sim.  Then I converted usage, and added a console.log() to double check the new code was being run in the sim.  I removed the console.log before committing.  When committing, I referenced this issue URL so the change is easy to track.
